tests/models/mega/test_modeling_mega.py

```diff
@@ -615,6 +625,39 @@ def test_model_from_pretrained(self):
             model = MegaModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
 
+    @unittest.skip(reason="Does not work on the tiny model as we keep hitting edge cases.")
+    def test_cpu_offload(self):
+        super().test_cpu_offload()
+
+    @unittest.skip(reason="Does not work on the tiny model as we keep hitting edge cases.")
+    def test_disk_offload(self):
+        super().test_disk_offload()
+
+    @unittest.skip(reason="Does not work on the tiny model as we keep hitting edge cases.")
+    def test_model_parallelism(self):
+        super().test_model_parallelism()
+
+    @unittest.skip(
+        reason=(
+            "Calling `self.attention_function` in `MegaMovingAverageGatedAttention.forward` changes the submodules on "
+            "device 1 to device 0 (also changes `requires_grad`). No idea how this could happen for now."
+        )
+    )
+    def test_multi_gpu_data_parallel_forward(self):
+        super().test_multi_gpu_data_parallel_forward()
+
+    @unittest.skip(reason="Tracing of the dynamically computed `MegaMultiDimensionDampedEma._kernel` doesn't work.")
+    def test_torchscript_simple(self):
+        super().test_torchscript_simple()
+
+    @unittest.skip(reason="Tracing of the dynamically computed `MegaMultiDimensionDampedEma._kernel` doesn't work.")
+    def test_torchscript_output_hidden_state(self):
+        super().test_torchscript_output_hidden_state()
+
+    @unittest.skip(reason="Tracing of the dynamically computed `MegaMultiDimensionDampedEma._kernel` doesn't work.")
+    def test_torchscript_output_attentions(self):
+        super().test_torchscript_output_attentions()
```

Comment:
`self._kernel` of `MegaMultiDimensionDampedEma` is computed dynamically. `torchscript` complains it can't trace a tensor with required grad.

(if input length is changed, it will be re-computed, and the result size will be different - `torchscript` won't allow this anyway)

Reply:
We can probably set a max input length and pre-compute, then subsample?

Reply:
We can do so, but it still won't work, as the first thing `torchscript` complains is about it can't trace a tensor (assigned to the module, here is `self._kernel`) that requires gradient. Even if we pre-compute and set it, it still won't be able to trace.

Reply:
(in fact, in our case, it's already set when  `model(main_input)` is executed, see below)

```python
model(main_input)
traced_model = torch.jit.trace(model, main_input)
```